addons/binding/org.openhab.binding.harmonyhub/src/main/java/org/openhab/binding/harmonyhub/handler/HarmonyHubHandler.java

```diff
@@ -124,14 +124,15 @@ public void handleCommand(ChannelUID channelUID, Command command) {
 
     @Override
     public void initialize() {
+        buttonExecutor = Executors.newSingleThreadScheduledExecutor();
         cancelRetry();
         connect();
     }
 
     @Override
     public void dispose() {
-        listeners.clear();
         buttonExecutor.shutdownNow();
```

Comment:
The pure goal of 'dispose/initialize sequences' is to safely handle configuration updates. 

Note that dispose is not some kind of destructor and was as far as I know never intended like that. Maybe you should not shutdown the executor but the thread(s) instead, would that make more sense? @sjka or am I missing a / the point here?

Just to be sure, the reason it is a single threaded scheduler and we cannot use the parent one is because of the restriction to send only one command at a time to a single harmony hub?